extensions/undertow/deployment/src/main/java/io/quarkus/undertow/deployment/UndertowBuildStep.java

```diff
@@ -194,6 +187,15 @@ public void kubernetes(HttpConfig config, BuildProducer<KubernetesPortBuildItem>
         portProducer.produce(new KubernetesPortBuildItem(config.port, "http"));
     }
 
+    /**
+     * Register the undertow-handlers.conf file
+     */
+    @BuildStep
+    @Record(STATIC_INIT)
+    public ServletExtensionBuildItem registerUndertowHandlersConf(UndertowDeploymentTemplate template) {
+        return new ServletExtensionBuildItem(template.setupUndertowHandlers());
```

Comment:
This should be checking for the undertow-handlers.conf file here, and only creating the extension if it is required.